server/supernova/sc/sc_osc_handler.cpp

```diff
@@ -2615,15 +2618,12 @@ template <bool realtime> void handle_b_getn(ReceivedMessage const& msg, endpoint
 }
 
 
-template <bool realtime> void handle_b_gen(ReceivedMessage const& msg, size_t msg_size, endpoint_ptr endpoint) {
-    movable_array<char> cmd(msg_size, msg.AddressPattern());
+template <bool realtime>
+void handle_b_gen(ReceivedMessage const& msg, size_t msg_size, endpoint_ptr endpoint, int skip_bytes) {
```

Comment:
This code has been there before. `TypeTags()` alone is not enough, because we also need to adjust the size, so we would have to do `msg_size - (msg.TypeTags() - msg.AddressPattern())` in all methods which need to instantiate a `sc_msg_iter` - which is redundant because we already know the offset beforehand... Therefore I would prefer to keep it as it is/was.

Reply:
no, please change it, it's much more expressive. you can write a helper function if you need to.

Reply:
I really don't like to recompute a value we already know. Can't I just add a comment, like `// skip address part`? Or rename `skip_bytes` to a more expressive `address_pattern_size`? I think with the latter it should be really clear what's going on.

Reply:
no, please change it. as @joshpar  pointed out in a previous review it requires magic numbers and per-usage comments to explain what the current implementation does. that is absolutely not good design. we are talking about subtracting two pointers here, it's really not that much of a sacrifice. if it were, we'd be writing this in C :)

Reply:
on the other hand, it seems that other people also found this part of the code confusing, so I just go ahead and skip the address pattern in the method handler. the runtime cost is negliglible.

Reply:
I've changed my mind in the meantime :-) I saw you comment only now

Reply:
@brianlheim actually, `TypeTags()` returns a pointer to the first type tag, not to the start of the type tag string (which starts with `,`), so I need to write `msg_size - (msg.TypeTags() - msg.AddressPattern() + 1); `. I don't think this is much clearer at all, but I just go ahead and do it.

Reply:
good catch, thank you for that!

Reply:
found out because my tests failed :-)